pkg/gui/controllers/commits_files_controller.go

```diff
@@ -231,21 +233,26 @@ func (self *CommitFilesController) toggleForPatch(node *filetree.CommitFileNode)
 				}
 			}
 
-			// if there is any file that hasn't been fully added we'll fully add everything,
-			// otherwise we'll remove everything
-			adding := node.SomeFile(func(file *models.CommitFile) bool {
-				return self.c.Git().Patch.PatchBuilder.GetFileStatus(file.Name, self.context().GetRef().RefName()) != patch.WHOLE
-			})
+			selectedNodes = normalisedSelectedCommitNodes(selectedNodes)
+
+			for _, node := range selectedNodes {
 
-			err := node.ForEachFile(func(file *models.CommitFile) error {
-				if adding {
-					return self.c.Git().Patch.PatchBuilder.AddFileWhole(file.Name)
-				} else {
-					return self.c.Git().Patch.PatchBuilder.RemoveFile(file.Name)
+				// if there is any file that hasn't been fully added we'll fully add everything,
+				// otherwise we'll remove everything
+				adding := node.SomeFile(func(file *models.CommitFile) bool {
```

Comment:
This belongs outside the loop and should consider all selected nodes: we don't want to be adding files for some nodes and removing others

Reply:
Ah! I see what you mean. I think I got this behavior fixed in 2c93be4f68d84100cac0a294b89cef66ac63b24d. I knocked together another test just to make sure the behavior is what we want. I'm happy to remove it later, but figured I'd leave it in just for now.